light_prop/propagation.py

```diff
@@ -14,45 +14,39 @@
 from keras.models import backend as K
 import tensorflow as tf
 
-from light_prop.calculations import h, gaussian, lens
+from light_prop.calculations import h
 from light_prop.propagation_params import PropagationParams
+from light_prop.propagation_results import PropagationResult
 
 
 class BasePropagation:
     def __int__(self, propagation_params: PropagationParams):
         self.params = propagation_params
 
-    def propagate(self):
+    def propagate(self, propagation_input):
         logging.info("Calculating propagation")
-        field_distribution = self.get_field_distribution()
+        field_distribution = self.get_field_distribution(propagation_input)
         field_modifier = self.get_field_modifier()
-        return self.calculate_propagation(field_distribution, field_modifier)
+        output = self.reshape_output(self.calculate_propagation(field_distribution, field_modifier))
+        return PropagationResult(output)
```

Comment:
I'd expect `LightField` here

Reply:
And in the input as well!

Reply:
yea but since there are not explicit types in python I didnt see that :D possible topic for validation